discovery/consul/consul.go

```diff
@@ -203,21 +233,46 @@ func (d *Discovery) shouldWatch(name string) bool {
 	return false
 }
 
+// shouldWatch returns whether the service of the given name should be watched based on its tags.
+func (d *Discovery) shouldWatchFromTags(tags []string) bool {
```

Comment:
This looks like filtering on our side at a first glance, which wouldn't be kosher.

It should be made clear in the comments that this is an optimisation on top of the option passed to the Service call, not a duplication of what should be in relabelling. We're going to get questions for example as to why only one tag is supported.

Reply:
I updated the documentation and comments. Hope it's more clear now.